### PR TITLE
Use open with context manager

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

File: hacking/test-module.py [hunk 1]
```diff
@@ -90,12 +90,11 @@ def parse():
 def write_argsfile(argstring, json=False):
     """ Write args to a file for old-style module's use. """
     argspath = os.path.expanduser("~/.ansible_test_module_arguments")
-    argsfile = open(argspath, 'w')
     if json:
         args = parse_kv(argstring)
         argstring = jsonify(args)
-    argsfile.write(argstring)
-    argsfile.close()
+    with open(argspath, 'w') as argsfile:
```
Comment (webknjaz) [inline hacking/test-module.py]: For such cases I'd even use `pathlib.Path.write_text()`.

File: test/units/modules/test_known_hosts.py [hunk 2]
```diff
@@ -14,11 +14,13 @@
...
-        tmp_file.write(to_bytes(content))
-        tmp_file.close()
-        self.addCleanup(os.unlink, tmp_file.name)
-        return tmp_file.name
+        tmp_filename = ''
+        with tempfile.NamedTemporaryFile(prefix='ansible-test-', suffix='-known_hosts', delete=False) as tmp_file:
+            tmp_file.write(to_bytes(content))
+            tmp_filename = tmp_file.name
+
+        self.addCleanup(os.unlink, tmp_filename)
```
Comment (webknjaz) [inline test/units/modules/test_known_hosts.py]: I think that the CM would delete the file prematurely in this specific instance. So the change is likely undesired here.

**Comment by Pritpal1999** [comment]
hi @Akasurde , I'm conducting research on pull request activities in open-source projects. Your expertise would be invaluable to my study. I need your insights on interpretation of results that could greatly enhance the findings. If you're open to it, please let me know!

**Review by s-hertel — APPROVED** [review]

**Pull request merged**